packages/fireworks/src/fireworks-provider.ts

```diff
@@ -0,0 +1,165 @@
+import {
+  OpenAICompatibleChatLanguageModel,
+  OpenAICompatibleCompletionLanguageModel,
+  OpenAICompatibleEmbeddingModel,
+  ProviderErrorStructure,
+} from '@ai-sdk/openai-compatible';
+import { EmbeddingModelV1, LanguageModelV1 } from '@ai-sdk/provider';
+import {
+  FetchFunction,
+  loadApiKey,
+  withoutTrailingSlash,
+} from '@ai-sdk/provider-utils';
+import { z } from 'zod';
+import {
+  FireworksChatModelId,
+  FireworksChatSettings,
+} from './fireworks-chat-settings';
+import {
+  FireworksCompletionModelId,
+  FireworksCompletionSettings,
+} from './fireworks-completion-settings';
+import {
+  FireworksEmbeddingModelId,
+  FireworksEmbeddingSettings,
+} from './fireworks-embedding-settings';
+
+type FireworksErrorData = z.infer<typeof fireworksErrorSchema>;
+
+const fireworksErrorSchema = z.object({
+  error: z.string().optional(),
+});
+
+const fireworksErrorStructure: ProviderErrorStructure<FireworksErrorData> = {
+  errorSchema: fireworksErrorSchema,
+  errorToMessage: data => data.error ?? 'unknown error',
```

Comment:
default was something that was caught by type inference. do we know if error is optional?

Reply:
I don't see a spec on it, I marked it optional presumptively, let's try making it required.

The embedding response as mentioned elsewhere is just a raw string with no keys or JSON formatting.

I expect we may make changes here in time.